presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -92,6 +92,95 @@ Additionally, the relations within a ``WITH`` clause can chain::
     relation is used. This means that if the relation is used more than once and the query
     is non-deterministic, the results may be different each time.
 
+WITH RECURSIVE Clause
+---------------------
+
+The ``WITH RECURSIVE`` clause is a variant of the ``WITH`` clause. It can define
+a list of queries to process, including recursive processing of suitable
+queries.
+
+.. warning::
+
+    This feature is experimental only. Proceed to use it only if you understand
+    potential query failures and the impact of the recursion processing on your
+    workload.
+
+.. code-block:: none
+
+   WITH RECURSIVE T(n) AS (
+    <recursion base relation>
+    UNION (ALL | DISTINCT)
+    <recursion step relation: SELECT  FROM ... T...>
+    )
+    ... SELECT ... FROM T
+
+A recursive ``WITH``-query must be shaped as a ``UNON`` of two relations. The
+first relation is called the *recursion base*, and the second relation is called
+the *recursion step*. Presto supports recursive ``WITH``-queries with a single
+recursive reference to a ``WITH``-query from within the query. The name ``T`` of
+the query ``T`` can be mentioned once in the ``FROM`` clause of the recursion
+step relation.
+
+The following listing shows a simple example, that displays a commonly used
+form of only one query in the list:
+
+.. code-block:: none
+
+    WITH RECURSIVE t(n) AS (
+        VALUES (1)
+        UNION ALL
+        SELECT n+1 FROM t WHERE n < 4
+    )
+    SELECT sum(n) FROM t;
+
+In the preceding query the simple assignment ``VALUES (1)`` defines the
+recursion base relation. ``SELECT n+1 FROM t WHERE n < 4`` defines the recursion
+step relation. The recursion processing performs these steps:
```

Comment:
This list lacks "step 0" which is computing the recursion base (`1`).
Maybe add this in the beginning of the list:
```
- recursion base yields ``1``
```

Reply:
added